### DISCUSSION TITLE
Getting Succubus Dungeon mod to detect DFHack in 50.13

### DISCUSSION THREAD

**Discussion by SporkofDeath** [DFHack/dfhack#4608] [Q&A] [answered]
I should start by saying I'm not a coder.  I've been playing the Succubus mod.  I started out playing it using the exact DF version that it says it's for on DFFD (50.09) but the 50.09 version of DFHack doesn't have the designation dimensions indicator feature on the HUD, which made creating all sorts of areas extremely difficult and frustrating using mouse control.  In order to be sure of the size I'd have to manually count everything out, so I thought I'd try it with a more recent version of the game and DFHack, and it seems to work just fine, except for one issue which I only realized after having played my current fort for a while.  It doesn't seem to realize that DFHack is enabled, and consequently the features that require DFHack are disabled.  I was wondering if anyone here potentially offer some insights to help me out.  I asked the mod creator on the steam discussion forum but they don't seem to be super active, so I figured I'd hedge my bets and ask here too.

**Comment by myk002** [comment]
I have zero knowledge of the Succubus Dungeon mod, but it should be fairly clear from looking at the source where it detect/depends on DFHack. do you have a link?

**Reply by silverwing235** [reply]
Apologies for jumping both of your guns, but this inspired me to go on a little delve into the Workshop, and, well, this might be it? (https://steamcommunity.com/sharedfiles/filedetails/?id=2950544248&searchtext=Succubus+Dungeon)

**Reply by SporkofDeath** [reply]
That's correct, though I'm using the DFFD download here as I don't have the steam version: https://dffd.bay12games.com/file.php?id=7492

**Reply by myk002** [reply, marked as answer]
The version on Steam appears to be more recent than the one on DFFD, and it has this in the changelog:

* The dfhack workshops are fixed for 50.11-r6

You can download that version even if you don't have Steam via a workshop downloader. I worked with Tarn to make the DF Steam Workshop non-exclusive to those who have purchased DF via Steam.

**Reply by myk002** [reply]
btw if you're upgrading from 50.09, you might be interested in all the improvements to DFHack since then: https://docs.dfhack.org/en/latest/docs/NEWS.html#changelog

**Reply by SporkofDeath** [reply]
Oh, interesting!  Thanks!  I was actually only using 50.09 specifically for that mod.  I'm used to more recent versions of DFHack, which is why I missed the designation dimensions info on the UI so much when first playing the mod in 50.09.    I guess that answers my questions for the most part as 50.11 has that feature.  I can just use DF 50.11.  I am curious if my current 50.13 fort is salvageable.

**Reply by SporkofDeath** [reply]
Uh, an addendum, do you have any suggestions for reliable workshop downloader?  I've never used one before and it seems like one of those things where I should be careful what I download.  Is the WorkshopDL one here on GitHub good?

**Reply by myk002** [reply]
The Succubus Dungeon changelog says it was *fixed* for DF 50.11, but I suspect it will work just fine in the current versions of DF and DFHack.

I'm not sure if I can recommend a downloader other than the official steamcmd client. There are instructions for how to login anonymously and download workshop mods here: https://www.reddit.com/r/DivinityOriginalSin/comments/73y9g4/how_to_download_steam_workshop_mods_for_the_gog/

**Reply by SporkofDeath** [reply]
Ok, I think I got it mostly working.  Looks like it works now when generating a new world, but not when loading an old fort, or a world generated with the old version of the mod but current version of DF, even if I haven't played in it yet.  Oh well, guess I'm glad I only had that fort running for like 10 years before I found out it wasn't working fully.  Thanks so much for all the help!  Thank goodness for my 90s kid DOS experience.

**Comment by silverwing235** [comment]
> btw if you're upgrading from 50.09, you might be interested in all the improvements to DFHack since then: https://docs.dfhack.org/en/latest/docs/NEWS.html#changelog
thx, am on adv mode beta anyhow :D
